Fix bindings security checks of RemoteDOMWindow property access

achristensen07 opened the pull request:
#### 4bd79fbe48607802bb5a86ff8b991e8ca457c825
<pre>
Fix bindings security checks of RemoteDOMWindow property access
<a href="https://bugs.webkit.org/show_bug.cgi?id=272632">https://bugs.webkit.org/show_bug.cgi?id=272632</a>
<a href="https://rdar.apple.com/126419644">rdar://126419644</a>

Reviewed by Charlie Wolfe.

In order to make more exotic ways of using JS to access properties on DOMWindows,
we need to have one IDL for LocalDOMWindow and RemoteDOMWindow that chooses which
object type to access properties of at runtime.  I introduced DOMWindowType so that
determination can be a well-predicted branch instead of a virtual function call.

This makes http/tests/security/cross-origin-window-property-access.html work with
site isolation enabled.  There is a bit of cleanup to do, but this PR is big enough.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/cache/WindowOrWorkerGlobalScopeCaches.cpp:
* Source/WebCore/Modules/cache/WindowOrWorkerGlobalScopeCaches.h:
* Source/WebCore/Modules/cookie-store/LocalDOMWindow+CookieStore.idl:
* Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.cpp:
(WebCore::WindowOrWorkerGlobalScopeFetch::fetch):
* Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.h:
* Source/WebCore/Modules/indexeddb/WindowOrWorkerGlobalScopeIndexedDatabase.cpp:
(WebCore::WindowOrWorkerGlobalScopeIndexedDatabase::indexedDB):
* Source/WebCore/Modules/indexeddb/WindowOrWorkerGlobalScopeIndexedDatabase.h:
* Source/WebCore/Modules/speech/LocalDOMWindow+SpeechSynthesis.idl:
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.cpp:
(WebCore::LocalDOMWindowSpeechSynthesis::LocalDOMWindowSpeechSynthesis):
(WebCore::LocalDOMWindowSpeechSynthesis::from):
(WebCore::LocalDOMWindowSpeechSynthesis::speechSynthesis):
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.h:
* Source/WebCore/Modules/webdatabase/Database.cpp:
* Source/WebCore/Modules/webdatabase/LocalDOMWindow+WebDatabase.idl:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/CommonVM.cpp:
(WebCore::lexicalFrameFromCommonVM):
* Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
(WebCore::JSCustomElementInterface::upgradeElement):
(WebCore::JSCustomElementInterface::invokeCallback):
* Source/WebCore/bindings/js/JSDOMBindingSecurity.cpp:
(WebCore::remoteFrameAccessError):
(WebCore::reportErrorAccessingRemoteFrame):
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
* Source/WebCore/bindings/js/JSDOMBindingSecurity.h:
* Source/WebCore/bindings/js/JSDOMBindingSecurityInlines.h:
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
* Source/WebCore/bindings/js/JSDOMExceptionHandling.cpp:
(WebCore::reportException):
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::JSDOMGlobalObject::scriptExecutionContext const):
(WebCore::scriptModuleLoader):
(WebCore::toJSDOMGlobalObject):
* Source/WebCore/bindings/js/JSDOMPromise.cpp:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::JSDOMWindowBase):
(WebCore::JSDOMWindowBase::finishCreation):
(WebCore::JSDOMWindowBase::updateDocument):
(WebCore::JSDOMWindowBase::scriptExecutionContext const):
(WebCore::JSDOMWindowBase::printErrorMessage const):
(WebCore::JSDOMWindowBase::currentScriptExecutionOwner):
(WebCore::JSDOMWindowBase::reportViolationForUnsafeEval):
(WebCore::toJS):
(WebCore::toJSDOMWindow):
(WebCore::incumbentDOMWindow):
(WebCore::activeDOMWindow):
(WebCore::firstDOMWindow):
(WebCore::legacyActiveDOMWindowForAccessor):
(WebCore::toJSLocalDOMWindow): Deleted.
* Source/WebCore/bindings/js/JSDOMWindowBase.h:
(WebCore::toJS):
(WebCore::toJSDOMWindow):
(WebCore::toJSLocalDOMWindow): Deleted.
* Source/WebCore/bindings/js/JSDOMWindowCustom.cpp: Added.
(WebCore::JSDOMWindow::visitAdditionalChildren):
(WebCore::JSC_DEFINE_CUSTOM_GETTER):
(WebCore::allowsLegacyExpandoIndexedProperties):
(WebCore::jsDOMWindowGetOwnPropertySlotRestrictedAccess):
(WebCore::handleCommonCrossOriginProperties):
(WebCore::JSDOMWindow::getOwnPropertySlot):
(WebCore::JSDOMWindow::getOwnPropertySlotByIndex):
(WebCore::JSDOMWindow::put):
(WebCore::JSDOMWindow::putByIndex):
(WebCore::JSDOMWindow::deleteProperty):
(WebCore::JSDOMWindow::deletePropertyByIndex):
(WebCore::JSDOMWindow::analyzeHeap):
(WebCore::addCrossOriginPropertyNames):
(WebCore::addCrossOriginOwnPropertyNames):
(WebCore::addScopedChildrenIndexes):
(WebCore::JSDOMWindow::getOwnPropertyNames):
(WebCore::JSDOMWindow::defineOwnProperty):
(WebCore::JSDOMWindow::getPrototype):
(WebCore::JSDOMWindow::preventExtensions):
(WebCore::JSDOMWindow::event const):
(WebCore::DialogHandler::DialogHandler):
(WebCore::DialogHandler::dialogCreated):
(WebCore::DialogHandler::returnValue const):
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSDOMWindow::queueMicrotask):
(WebCore::JSDOMWindow::toWrapped):
(WebCore::JSDOMWindow::setOpener):
(WebCore::JSDOMWindow::self const):
(WebCore::JSDOMWindow::window const):
(WebCore::JSDOMWindow::frames const):
(WebCore::jsDOMWindowInstanceFunction_openDatabaseBody):
(WebCore::JSDOMWindow::openDatabase const):
(WebCore::JSDOMWindow::setOpenDatabase):
(WebCore::mainWorldGlobalObject):
* Source/WebCore/bindings/js/JSDOMWindowCustom.h: Renamed from Source/WebCore/bindings/js/JSLocalDOMWindowCustom.h.
(WebCore::asJSDOMWindow):
(WebCore::mainWorldGlobalObject):
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::JSDOMWindowProperties::getOwnPropertySlot):
* Source/WebCore/bindings/js/JSDOMWindowProperties.h:
* Source/WebCore/bindings/js/JSDOMWrapper.cpp:
(WebCore::JSDOMObject::JSDOMObject):
* Source/WebCore/bindings/js/JSDocumentCustom.cpp:
(WebCore::cachedDocumentWrapper):
* Source/WebCore/bindings/js/JSErrorHandler.cpp:
(WebCore::JSErrorHandler::handleEvent):
* Source/WebCore/bindings/js/JSEventListener.cpp:
(WebCore::JSEventListener::handleEvent):
* Source/WebCore/bindings/js/JSEventListener.h:
(WebCore::windowEventHandlerAttribute):
(WebCore::setWindowEventHandlerAttribute):
* Source/WebCore/bindings/js/JSEventTargetCustom.cpp:
(WebCore::JSEventTarget::toWrapped):
(WebCore::jsEventTargetCast):
* Source/WebCore/bindings/js/JSEventTargetCustom.h:
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
(WebCore::JSLazyEventListener::initializeJSFunction const):
(WebCore::JSLazyEventListener::create):
* Source/WebCore/bindings/js/JSLocationCustom.cpp:
(WebCore::getOwnPropertySlotCommon):
* Source/WebCore/bindings/js/JSMessageEventCustom.cpp:
* Source/WebCore/bindings/js/JSNodeCustom.cpp:
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.cpp: Removed.
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.h: Removed.
* Source/WebCore/bindings/js/JSRemoteDOMWindowCustom.cpp: Removed.
* Source/WebCore/bindings/js/JSWindowProxy.cpp:
(WebCore::JSWindowProxy::setWindow):
(WebCore::JSWindowProxy::wrapped const):
* Source/WebCore/bindings/js/JSWindowProxy.h:
* Source/WebCore/bindings/js/JSXMLDocumentCustom.cpp:
* Source/WebCore/bindings/js/ScheduledAction.cpp:
(WebCore::ScheduledAction::execute):
* Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
(WebCore::ScriptCachedFrameData::ScriptCachedFrameData):
* Source/WebCore/bindings/js/ScriptCachedFrameData.h:
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::initScriptForWindowProxy):
(WebCore::ScriptController::updateDocument):
* Source/WebCore/bindings/js/ScriptControllerMac.mm:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSHeapData::JSHeapData):
(WebCore::JSVMClientData::overrideSourceURL const):
* Source/WebCore/bindings/js/WebCoreJSClientData.h:
* Source/WebCore/bindings/js/WindowProxy.cpp:
(WebCore::WindowProxy::setDOMWindow):
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(IsDOMGlobalObject):
(GenerateHeader):
(GenerateOverloadDispatcher):
(GenerateRuntimeEnableConditionalString):
(GenerateImplementation):
(GenerateAttributeGetterBodyDefinition):
(GenerateAttributeSetterBodyDefinition):
(GenerateOperationBodyDefinition):
* Source/WebCore/bridge/objc/WebScriptObject.mm:
(WebCore::addExceptionToConsole):
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
* Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.cpp:
(WebCore::WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes):
* Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.h:
* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::getContext):
* Source/WebCore/inspector/InspectorController.cpp:
(WebCore::InspectorController::canAccessInspectedScriptState const):
* Source/WebCore/inspector/PageDebugger.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
* Source/WebCore/inspector/agents/page/PageAuditAgent.cpp:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::DOMWindow):
(WebCore::DOMWindow::name const):
(WebCore::DOMWindow::setName):
(WebCore::DOMWindow::status const):
(WebCore::DOMWindow::setStatus):
(WebCore::DOMWindow::length const):
(WebCore::DOMWindow::documentIfLocal):
(WebCore:: const):
(WebCore::DOMWindow::outerHeight const):
(WebCore::DOMWindow::outerWidth const):
(WebCore::DOMWindow::innerHeight const):
(WebCore::DOMWindow::innerWidth const):
(WebCore::DOMWindow::screenX const):
(WebCore::DOMWindow::screenY const):
(WebCore::DOMWindow::screenLeft const):
(WebCore::DOMWindow::screenTop const):
(WebCore::DOMWindow::scrollX const):
(WebCore::DOMWindow::scrollY const):
(WebCore::DOMWindow::offscreenBuffering const):
(WebCore::DOMWindow::devicePixelRatio const):
(WebCore::DOMWindow::origin const):
(WebCore::DOMWindow::isSecureContext const):
(WebCore::DOMWindow::crossOriginIsolated const):
(WebCore::DOMWindow::focus):
(WebCore::DOMWindow::blur):
(WebCore::DOMWindow::print):
(WebCore::DOMWindow::stop):
(WebCore::DOMWindow::postMessage):
(WebCore::DOMWindow::getComputedStyle const):
(WebCore::DOMWindow::matchMedia):
(WebCore::DOMWindow::open):
(WebCore::DOMWindow::alert):
(WebCore::DOMWindow::confirmForBindings):
(WebCore::DOMWindow::prompt):
(WebCore::DOMWindow::captureEvents):
(WebCore::DOMWindow::releaseEvents):
(WebCore::DOMWindow::find const):
(WebCore::DOMWindow::requestAnimationFrame):
(WebCore::DOMWindow::webkitRequestAnimationFrame):
(WebCore::DOMWindow::cancelAnimationFrame):
(WebCore::DOMWindow::requestIdleCallback):
(WebCore::DOMWindow::cancelIdleCallback):
(WebCore::DOMWindow::createImageBitmap):
(WebCore::DOMWindow::getMatchedCSSRules const):
(WebCore::DOMWindow::webkitConvertPointFromPageToNode const):
(WebCore::DOMWindow::webkitConvertPointFromNodeToPage const):
(WebCore::DOMWindow::collectMatchingElementsInFlatTree):
(WebCore::DOMWindow::matchingElementInFlatTree):
(WebCore::DOMWindow::scrollBy const):
(WebCore::DOMWindow::scrollTo const):
(WebCore::DOMWindow::moveBy const):
(WebCore::DOMWindow::moveTo const):
(WebCore::DOMWindow::resizeBy const):
(WebCore::DOMWindow::resizeTo const):
(WebCore::DOMWindow::setTimeout):
(WebCore::DOMWindow::clearTimeout):
(WebCore::DOMWindow::setInterval):
(WebCore::DOMWindow::clearInterval):
(WebCore::DOMWindow::reportError):
(WebCore::DOMWindow::structuredClone):
(WebCore::DOMWindow::btoa):
(WebCore::DOMWindow::atob):
* Source/WebCore/page/DOMWindow.h:
(WebCore::DOMWindow::isLocalDOMWindow const):
(WebCore::DOMWindow::isRemoteDOMWindow const):
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::takeWindowProxyFrom):
* Source/WebCore/page/LocalDOMWindow+CSSOM.idl:
* Source/WebCore/page/LocalDOMWindow+CSSOMView.idl:
* Source/WebCore/page/LocalDOMWindow+Compat.idl:
* Source/WebCore/page/LocalDOMWindow+DeviceMotion.idl:
* Source/WebCore/page/LocalDOMWindow+DeviceOrientation.idl:
* Source/WebCore/page/LocalDOMWindow+RequestIdleCallback.idl:
* Source/WebCore/page/LocalDOMWindow+Selection.idl:
* Source/WebCore/page/LocalDOMWindow+VisualViewport.idl:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::LocalDOMWindow):
* Source/WebCore/page/LocalDOMWindow.h:
(isType):
* Source/WebCore/page/LocalDOMWindow.idl: Removed.
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::fromJSContext):
(WebCore::LocalFrame::contentFrameFromWindowOrFrameElement):
* Source/WebCore/page/PerformanceLogging.cpp:
* Source/WebCore/page/RemoteDOMWindow.cpp:
(WebCore::RemoteDOMWindow::RemoteDOMWindow):
(WebCore::RemoteDOMWindow::setOpener):
* Source/WebCore/page/RemoteDOMWindow.h:
* Source/WebCore/page/RemoteDOMWindow.idl: Removed.
* Source/WebCore/page/ResourceUsageThread.cpp:
* Source/WebCore/page/cocoa/ResourceUsageOverlayCocoa.mm:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::windowLocationHost):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebCore/xml/XMLHttpRequest.cpp:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::reportException):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:
* Source/WebKitLegacy/mac/Misc/WebCoreStatistics.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _stringByEvaluatingJavaScriptFromString:withGlobalObject:inScriptWorld:]):
* Source/WebKitLegacy/mac/WebView/WebScriptDebugger.mm:
(toWebFrame):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView _reportException:inContext:]):

Canonical link: <a href="https://commits.webkit.org/277560@main">https://commits.webkit.org/277560@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0f0250e45a29dfdc2ec30cc15265eebc37170982

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47980 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27183 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50890 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50661 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44033 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50287 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33063 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24675 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39037 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48562 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24857 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20334 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22335 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42695 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6027 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/41273 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/44321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52555 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/47472 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23023 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19374 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46350 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24292 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41664 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45395 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25087 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/54969 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6804 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24014 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11305 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
```diff
@@ -22,7 +22,7 @@
 
 #include "CommonVM.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/JSDOMPromise.cpp:
```diff
@@ -26,7 +26,7 @@
 #include "JSDOMPromise.h"
 
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe replied:
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
```diff
@@ -29,7 +29,7 @@
 #include "JSDOMExceptionHandling.h"
 #include "JSDOMPromise.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
```diff
@@ -34,7 +34,7 @@
 namespace WebCore {
 
-class JSLocalDOMWindow;
+class JSDOMWindow;
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/JSDOMWindowProperties.h:
```diff
@@ -25,8 +25,8 @@
 
 #pragma once
 
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/JSMessageEventCustom.cpp:
```diff
@@ -35,8 +35,8 @@
 #include "JSBlob.h"
 #include "JSDOMBinding.h"
 #include "JSDOMConvert.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/JSNodeCustom.cpp:
```diff
@@ -40,13 +40,13 @@
 #include "JSCDATASection.h"
 #include "JSComment.h"
 #include "JSDOMBinding.h"
+#include "JSDOMWindowCustom.h"
```
This can be removed.

achristensen07 replied:
This is needed for mainWorldGlobalObject

charliewolfe commented on Source/WebCore/bindings/js/JSXMLDocumentCustom.cpp:
```diff
@@ -26,8 +26,8 @@
 #include "config.h"
 #include "JSXMLDocument.h"
 
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
```diff
@@ -52,8 +52,8 @@ ScriptCachedFrameData::ScriptCachedFrameData(LocalFrame& frame)
-        auto* window = jsCast<JSLocalDOMWindow*>(windowProxy->window());
-        m_windows.add(&windowProxy->world(), Strong<JSLocalDOMWindow>(window->vm(), window));
+        auto* window = jsCast<JSDOMWindow*>(windowProxy->window());
+        m_windows.add(&windowProxy->world(), Strong<JSDOMWindow>(window->vm(), window));
```
Pretty sure this one needs the JSDOMWindow include, otherwise will break the non-unified build.

charliewolfe commented on Source/WebCore/bindings/js/ScriptControllerMac.mm:
```diff
@@ -32,7 +32,7 @@
 #import "CommonVM.h"
 #import "FrameLoader.h"
-#import "JSLocalDOMWindow.h"
+#import "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/bindings/scripts/test/LocalDOMWindowConstructors.idl:
```diff
@@ -1,7 +1,7 @@
-partial interface LocalDOMWindow {
+partial interface DOMWindow {
```
Should this file name be renamed now?

achristensen07 replied:
This is part of the code generation.  I'm not going to change this in this PR.  Future cleanup.

charliewolfe commented on Source/WebCore/bindings/scripts/test/JS/JSDOMWindowConstructors.cpp:
```diff
@@ -0,0 +1,5 @@
+/*
+    This file is generated to inform build scripts that JSDOMWindowConstructors.h and
+    JSDOMWindowConstructors.cpp were created for DOMWindowConstructors.idl, and prevent the build
```
`DOMWindowConstructors.idl` doesn't exist (since the file name wasn't renamed).

achristensen07 replied:
ditto

charliewolfe commented on Source/WebCore/bindings/scripts/test/JS/JSDOMWindowConstructors.h:
```diff
@@ -0,0 +1,5 @@
+/*
+    This file is generated to inform build scripts that JSDOMWindowConstructors.h and
+    JSDOMWindowConstructors.cpp were created for DOMWindowConstructors.idl, and prevent the build
```
Same thing here, file doesn't exist.

achristensen07 replied:
ditto

charliewolfe commented on Source/WebCore/dom/ScriptExecutionContext.cpp:
```diff
@@ -41,7 +41,7 @@
 #include "JSDOMExceptionHandling.h"
 #include "JSDOMPromiseDeferred.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.h:
```diff
@@ -36,7 +36,7 @@ template<typename> class ExceptionOr;
 public:
     static ASCIILiteral workerGlobalSupplementName();
-    static TrustedTypePolicyFactory* trustedTypes(LocalDOMWindow&);
+    static TrustedTypePolicyFactory* trustedTypes(DOMWindow&);
```
The forward declaration above should be changed from `LocalDOMWindow` to `DOMWindow`.

charliewolfe commented on Source/WebCore/html/PDFDocument.cpp:
```diff
@@ -45,6 +45,7 @@
 #include "RawDataDocumentParser.h"
 #include "ScriptController.h"
 #include "Settings.h"
+#include <JavaScriptCore/ObjectConstructor.h>
```
Was this change intentional?

achristensen07 replied:
Yes.  This is needed to fix a build failure caused by removing a header included elsewhere.

charliewolfe commented on Source/WebCore/inspector/InspectorController.cpp:
```diff
@@ -55,9 +55,9 @@
 #include "InstrumentingAgents.h"
 #include "JSDOMBindingSecurity.h"
+#include "JSDOMWindow.h"
+#include "JSDOMWindowCustom.h"
```
This second import can be removed.

charliewolfe commented on Source/WebCore/inspector/InspectorController.cpp:
```diff
@@ -482,11 +482,11 @@ bool InspectorController::canAccessInspectedScriptState(JSC::JSGlobalObject* lex
         return false;
 
-    return BindingSecurity::shouldAllowAccessToDOMWindow(lexicalGlobalObject, inspectedWindow->wrapped(), DoNotReportSecurityError);
+    return BindingSecurity::shouldAllowAccessToDOMWindow(lexicalGlobalObject, &inspectedWindow->wrapped(), DoNotReportSecurityError);
```
I don't think this change is needed?

charliewolfe commented on Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
```diff
@@ -42,8 +42,8 @@
 #include "InspectorMemoryAgent.h"
 #include "InspectorPageAgent.h"
 #include "InstrumentingAgents.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/page/DOMWindow.idl:
```diff
@@ -10,20 +11,42 @@
  *    notice, this list of conditions and the following disclaimer in the
  *    documentation and/or other materials provided with the distribution.
  *
- * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
```
This seems like it shouldn't be changed. I think the old license was correct.

achristensen07 replied:
I'm moving LocalDOMWindow.idl to DOMWindow.idl and retaining the exact license of LocalDOMWindow.idl

charliewolfe commented on Source/WebCore/page/LocalDOMWindow+CSSOM.idl:
```diff
@@ -26,6 +26,6 @@
 
 // https://drafts.csswg.org/cssom/#extensions-to-the-window-interface
-partial interface LocalDOMWindow {
+partial interface DOMWindow {
```
Should this file be renamed now? Same with all the other `LocalDOMWindow+*.idl` files.

achristensen07 replied:
That's the cleanup I mentioned.  I'll rename in a future PR.

charliewolfe commented on Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.h:
```diff
@@ -35,7 +35,7 @@ class WorkerGlobalScope;
 class WindowOrWorkerGlobalScopeFetch {
 public:
-    static void fetch(LocalDOMWindow&, FetchRequest::Info&&, FetchRequest::Init&&, Ref<DeferredPromise>&&);
+    static void fetch(DOMWindow&, FetchRequest::Info&&, FetchRequest::Init&&, Ref<DeferredPromise>&&);
```
Forward declaration above should be changed to `DOMWindow` here too.

charliewolfe commented on Source/WebCore/Modules/webdatabase/Database.cpp:
```diff
@@ -39,7 +39,7 @@
 #include "DatabaseTracker.h"
 #include "Document.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/page/DOMWindow.cpp:
```diff
@@ -177,4 +183,733 @@ WindowProxy* DOMWindow::parent() const
+    auto* localThis = dynamicDowncast<LocalDOMWindow>(*this);
+    if (!localThis)
+        return Exception { ExceptionCode::SecurityError };
+    return localThis->name();
```
This file needs `#include "LocalDOMWindow.h"`

charliewolfe commented on Source/WebCore/page/PerformanceLogging.cpp:
```diff
@@ -30,7 +30,7 @@
 #include "Document.h"
 #include "FrameLoader.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/page/ResourceUsageThread.cpp:
```diff
@@ -29,7 +29,7 @@
 
 #include "CommonVM.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

charliewolfe commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -72,8 +72,8 @@
 		0562F9611573F88F0031CA16 /* PlatformLayer.h in Headers */ = {isa = PBXBuildFile; fileRef = 0562F9601573F88F0031CA16 /* PlatformLayer.h */; settings = {ATTRIBUTES = (Private, ); }; };
-		056A7AC8256C4F48002F9DDF /* GetHTMLOptions.h in Headers */ = {isa = PBXBuildFile; fileRef = 054E3A33256C3BD90072C5BF /* GetHTMLOptions.h */; settings = {ATTRIBUTES = (Private, ); }; };
 		056A7AC8256C4F48002F9DDC /* ShadowRootInit.h in Headers */ = {isa = PBXBuildFile; fileRef = 054E3A33256C3BD90072C5B9 /* ShadowRootInit.h */; settings = {ATTRIBUTES = (Private, ); }; };
+		056A7AC8256C4F48002F9DDF /* GetHTMLOptions.h in Headers */ = {isa = PBXBuildFile; fileRef = 054E3A33256C3BD90072C5BF /* GetHTMLOptions.h */; settings = {ATTRIBUTES = (Private, ); }; };
```
Some of these project file changes should probably be reverted?

charliewolfe commented on Source/WebCore/xml/XMLHttpRequest.cpp:
```diff
@@ -39,7 +39,7 @@
 #include "InspectorInstrumentation.h"
 #include "JSDOMBinding.h"
-#include "JSLocalDOMWindow.h"
+#include "JSDOMWindow.h"
```
This can be removed.

achristensen07 committed:
Fix bindings security checks of RemoteDOMWindow property access
https://bugs.webkit.org/show_bug.cgi?id=272632
rdar://126419644

Reviewed by Charlie Wolfe.

In order to make more exotic ways of using JS to access properties on DOMWindows,
we need to have one IDL for LocalDOMWindow and RemoteDOMWindow that chooses which
object type to access properties of at runtime.  I introduced DOMWindowType so that
determination can be a well-predicted branch instead of a virtual function call.

This makes http/tests/security/cross-origin-window-property-access.html work with
site isolation enabled.  There is a bit of cleanup to do, but this PR is big enough.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/cache/WindowOrWorkerGlobalScopeCaches.cpp:
* Source/WebCore/Modules/cache/WindowOrWorkerGlobalScopeCaches.h:
* Source/WebCore/Modules/cookie-store/LocalDOMWindow+CookieStore.idl:
* Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.cpp:
(WebCore::WindowOrWorkerGlobalScopeFetch::fetch):
* Source/WebCore/Modules/fetch/WindowOrWorkerGlobalScopeFetch.h:
* Source/WebCore/Modules/indexeddb/WindowOrWorkerGlobalScopeIndexedDatabase.cpp:
(WebCore::WindowOrWorkerGlobalScopeIndexedDatabase::indexedDB):
* Source/WebCore/Modules/indexeddb/WindowOrWorkerGlobalScopeIndexedDatabase.h:
* Source/WebCore/Modules/speech/LocalDOMWindow+SpeechSynthesis.idl:
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.cpp:
(WebCore::LocalDOMWindowSpeechSynthesis::LocalDOMWindowSpeechSynthesis):
(WebCore::LocalDOMWindowSpeechSynthesis::from):
(WebCore::LocalDOMWindowSpeechSynthesis::speechSynthesis):
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.h:
* Source/WebCore/Modules/webdatabase/Database.cpp:
* Source/WebCore/Modules/webdatabase/LocalDOMWindow+WebDatabase.idl:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/CommonVM.cpp:
(WebCore::lexicalFrameFromCommonVM):
* Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
(WebCore::JSCustomElementInterface::upgradeElement):
(WebCore::JSCustomElementInterface::invokeCallback):
* Source/WebCore/bindings/js/JSDOMBindingSecurity.cpp:
(WebCore::remoteFrameAccessError):
(WebCore::reportErrorAccessingRemoteFrame):
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
* Source/WebCore/bindings/js/JSDOMBindingSecurity.h:
* Source/WebCore/bindings/js/JSDOMBindingSecurityInlines.h:
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
* Source/WebCore/bindings/js/JSDOMExceptionHandling.cpp:
(WebCore::reportException):
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::JSDOMGlobalObject::scriptExecutionContext const):
(WebCore::scriptModuleLoader):
(WebCore::toJSDOMGlobalObject):
* Source/WebCore/bindings/js/JSDOMPromise.cpp:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::JSDOMWindowBase):
(WebCore::JSDOMWindowBase::finishCreation):
(WebCore::JSDOMWindowBase::updateDocument):
(WebCore::JSDOMWindowBase::scriptExecutionContext const):
(WebCore::JSDOMWindowBase::printErrorMessage const):
(WebCore::JSDOMWindowBase::currentScriptExecutionOwner):
(WebCore::JSDOMWindowBase::reportViolationForUnsafeEval):
(WebCore::toJS):
(WebCore::toJSDOMWindow):
(WebCore::incumbentDOMWindow):
(WebCore::activeDOMWindow):
(WebCore::firstDOMWindow):
(WebCore::legacyActiveDOMWindowForAccessor):
(WebCore::toJSLocalDOMWindow): Deleted.
* Source/WebCore/bindings/js/JSDOMWindowBase.h:
(WebCore::toJS):
(WebCore::toJSDOMWindow):
(WebCore::toJSLocalDOMWindow): Deleted.
* Source/WebCore/bindings/js/JSDOMWindowCustom.cpp: Added.
(WebCore::JSDOMWindow::visitAdditionalChildren):
(WebCore::JSC_DEFINE_CUSTOM_GETTER):
(WebCore::allowsLegacyExpandoIndexedProperties):
(WebCore::jsDOMWindowGetOwnPropertySlotRestrictedAccess):
(WebCore::handleCommonCrossOriginProperties):
(WebCore::JSDOMWindow::getOwnPropertySlot):
(WebCore::JSDOMWindow::getOwnPropertySlotByIndex):
(WebCore::JSDOMWindow::put):
(WebCore::JSDOMWindow::putByIndex):
(WebCore::JSDOMWindow::deleteProperty):
(WebCore::JSDOMWindow::deletePropertyByIndex):
(WebCore::JSDOMWindow::analyzeHeap):
(WebCore::addCrossOriginPropertyNames):
(WebCore::addCrossOriginOwnPropertyNames):
(WebCore::addScopedChildrenIndexes):
(WebCore::JSDOMWindow::getOwnPropertyNames):
(WebCore::JSDOMWindow::defineOwnProperty):
(WebCore::JSDOMWindow::getPrototype):
(WebCore::JSDOMWindow::preventExtensions):
(WebCore::JSDOMWindow::event const):
(WebCore::DialogHandler::DialogHandler):
(WebCore::DialogHandler::dialogCreated):
(WebCore::DialogHandler::returnValue const):
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSDOMWindow::queueMicrotask):
(WebCore::JSDOMWindow::toWrapped):
(WebCore::JSDOMWindow::setOpener):
(WebCore::JSDOMWindow::self const):
(WebCore::JSDOMWindow::window const):
(WebCore::JSDOMWindow::frames const):
(WebCore::jsDOMWindowInstanceFunction_openDatabaseBody):
(WebCore::JSDOMWindow::openDatabase const):
(WebCore::JSDOMWindow::setOpenDatabase):
(WebCore::mainWorldGlobalObject):
* Source/WebCore/bindings/js/JSDOMWindowCustom.h: Renamed from Source/WebCore/bindings/js/JSLocalDOMWindowCustom.h.
(WebCore::asJSDOMWindow):
(WebCore::mainWorldGlobalObject):
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::JSDOMWindowProperties::getOwnPropertySlot):
* Source/WebCore/bindings/js/JSDOMWindowProperties.h:
* Source/WebCore/bindings/js/JSDOMWrapper.cpp:
(WebCore::JSDOMObject::JSDOMObject):
* Source/WebCore/bindings/js/JSDocumentCustom.cpp:
(WebCore::cachedDocumentWrapper):
* Source/WebCore/bindings/js/JSErrorHandler.cpp:
(WebCore::JSErrorHandler::handleEvent):
* Source/WebCore/bindings/js/JSEventListener.cpp:
(WebCore::JSEventListener::handleEvent):
* Source/WebCore/bindings/js/JSEventListener.h:
(WebCore::windowEventHandlerAttribute):
(WebCore::setWindowEventHandlerAttribute):
* Source/WebCore/bindings/js/JSEventTargetCustom.cpp:
(WebCore::JSEventTarget::toWrapped):
(WebCore::jsEventTargetCast):
* Source/WebCore/bindings/js/JSEventTargetCustom.h:
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
(WebCore::JSLazyEventListener::initializeJSFunction const):
(WebCore::JSLazyEventListener::create):
* Source/WebCore/bindings/js/JSLocationCustom.cpp:
(WebCore::getOwnPropertySlotCommon):
* Source/WebCore/bindings/js/JSMessageEventCustom.cpp:
* Source/WebCore/bindings/js/JSNodeCustom.cpp:
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.cpp: Removed.
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.h: Removed.
* Source/WebCore/bindings/js/JSRemoteDOMWindowCustom.cpp: Removed.
* Source/WebCore/bindings/js/JSWindowProxy.cpp:
(WebCore::JSWindowProxy::setWindow):
(WebCore::JSWindowProxy::wrapped const):
* Source/WebCore/bindings/js/JSWindowProxy.h:
* Source/WebCore/bindings/js/JSXMLDocumentCustom.cpp:
* Source/WebCore/bindings/js/ScheduledAction.cpp:
(WebCore::ScheduledAction::execute):
* Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
(WebCore::ScriptCachedFrameData::ScriptCachedFrameData):
* Source/WebCore/bindings/js/ScriptCachedFrameData.h:
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::initScriptForWindowProxy):
(WebCore::ScriptController::updateDocument):
* Source/WebCore/bindings/js/ScriptControllerMac.mm:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSHeapData::JSHeapData):
(WebCore::JSVMClientData::overrideSourceURL const):
* Source/WebCore/bindings/js/WebCoreJSClientData.h:
* Source/WebCore/bindings/js/WindowProxy.cpp:
(WebCore::WindowProxy::setDOMWindow):
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(IsDOMGlobalObject):
(GenerateHeader):
(GenerateOverloadDispatcher):
(GenerateRuntimeEnableConditionalString):
(GenerateImplementation):
(GenerateAttributeGetterBodyDefinition):
(GenerateAttributeSetterBodyDefinition):
(GenerateOperationBodyDefinition):
* Source/WebCore/bridge/objc/WebScriptObject.mm:
(WebCore::addExceptionToConsole):
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
* Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.cpp:
(WebCore::WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes):
* Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.h:
* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::getContext):
* Source/WebCore/inspector/InspectorController.cpp:
(WebCore::InspectorController::canAccessInspectedScriptState const):
* Source/WebCore/inspector/PageDebugger.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
* Source/WebCore/inspector/agents/page/PageAuditAgent.cpp:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::DOMWindow):
(WebCore::DOMWindow::name const):
(WebCore::DOMWindow::setName):
(WebCore::DOMWindow::status const):
(WebCore::DOMWindow::setStatus):
(WebCore::DOMWindow::length const):
(WebCore::DOMWindow::documentIfLocal):
(WebCore:: const):
(WebCore::DOMWindow::outerHeight const):
(WebCore::DOMWindow::outerWidth const):
(WebCore::DOMWindow::innerHeight const):
(WebCore::DOMWindow::innerWidth const):
(WebCore::DOMWindow::screenX const):
(WebCore::DOMWindow::screenY const):
(WebCore::DOMWindow::screenLeft const):
(WebCore::DOMWindow::screenTop const):
(WebCore::DOMWindow::scrollX const):
(WebCore::DOMWindow::scrollY const):
(WebCore::DOMWindow::offscreenBuffering const):
(WebCore::DOMWindow::devicePixelRatio const):
(WebCore::DOMWindow::origin const):
(WebCore::DOMWindow::isSecureContext const):
(WebCore::DOMWindow::crossOriginIsolated const):
(WebCore::DOMWindow::focus):
(WebCore::DOMWindow::blur):
(WebCore::DOMWindow::print):
(WebCore::DOMWindow::stop):
(WebCore::DOMWindow::postMessage):
(WebCore::DOMWindow::getComputedStyle const):
(WebCore::DOMWindow::matchMedia):
(WebCore::DOMWindow::open):
(WebCore::DOMWindow::alert):
(WebCore::DOMWindow::confirmForBindings):
(WebCore::DOMWindow::prompt):
(WebCore::DOMWindow::captureEvents):
(WebCore::DOMWindow::releaseEvents):
(WebCore::DOMWindow::find const):
(WebCore::DOMWindow::requestAnimationFrame):
(WebCore::DOMWindow::webkitRequestAnimationFrame):
(WebCore::DOMWindow::cancelAnimationFrame):
(WebCore::DOMWindow::requestIdleCallback):
(WebCore::DOMWindow::cancelIdleCallback):
(WebCore::DOMWindow::createImageBitmap):
(WebCore::DOMWindow::getMatchedCSSRules const):
(WebCore::DOMWindow::webkitConvertPointFromPageToNode const):
(WebCore::DOMWindow::webkitConvertPointFromNodeToPage const):
(WebCore::DOMWindow::collectMatchingElementsInFlatTree):
(WebCore::DOMWindow::matchingElementInFlatTree):
(WebCore::DOMWindow::scrollBy const):
(WebCore::DOMWindow::scrollTo const):
(WebCore::DOMWindow::moveBy const):
(WebCore::DOMWindow::moveTo const):
(WebCore::DOMWindow::resizeBy const):
(WebCore::DOMWindow::resizeTo const):
(WebCore::DOMWindow::setTimeout):
(WebCore::DOMWindow::clearTimeout):
(WebCore::DOMWindow::setInterval):
(WebCore::DOMWindow::clearInterval):
(WebCore::DOMWindow::reportError):
(WebCore::DOMWindow::structuredClone):
(WebCore::DOMWindow::btoa):
(WebCore::DOMWindow::atob):
* Source/WebCore/page/DOMWindow.h:
(WebCore::DOMWindow::isLocalDOMWindow const):
(WebCore::DOMWindow::isRemoteDOMWindow const):
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::takeWindowProxyFrom):
* Source/WebCore/page/LocalDOMWindow+CSSOM.idl:
* Source/WebCore/page/LocalDOMWindow+CSSOMView.idl:
* Source/WebCore/page/LocalDOMWindow+Compat.idl:
* Source/WebCore/page/LocalDOMWindow+DeviceMotion.idl:
* Source/WebCore/page/LocalDOMWindow+DeviceOrientation.idl:
* Source/WebCore/page/LocalDOMWindow+RequestIdleCallback.idl:
* Source/WebCore/page/LocalDOMWindow+Selection.idl:
* Source/WebCore/page/LocalDOMWindow+VisualViewport.idl:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::LocalDOMWindow):
* Source/WebCore/page/LocalDOMWindow.h:
(isType):
* Source/WebCore/page/LocalDOMWindow.idl: Removed.
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::fromJSContext):
(WebCore::LocalFrame::contentFrameFromWindowOrFrameElement):
* Source/WebCore/page/PerformanceLogging.cpp:
* Source/WebCore/page/RemoteDOMWindow.cpp:
(WebCore::RemoteDOMWindow::RemoteDOMWindow):
(WebCore::RemoteDOMWindow::setOpener):
* Source/WebCore/page/RemoteDOMWindow.h:
* Source/WebCore/page/RemoteDOMWindow.idl: Removed.
* Source/WebCore/page/ResourceUsageThread.cpp:
* Source/WebCore/page/cocoa/ResourceUsageOverlayCocoa.mm:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::windowLocationHost):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebCore/xml/XMLHttpRequest.cpp:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::reportException):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:
* Source/WebKitLegacy/mac/Misc/WebCoreStatistics.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _stringByEvaluatingJavaScriptFromString:withGlobalObject:inScriptWorld:]):
* Source/WebKitLegacy/mac/WebView/WebScriptDebugger.mm:
(toWebFrame):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView _reportException:inContext:]):

Canonical link: https://commits.webkit.org/277560@main
